sdk/monitor/monitor-opentelemetry-exporter/src/platform/nodejs/baseSender.ts

```diff
@@ -90,16 +112,29 @@ export abstract class BaseSender {
           }, this.batchSendRetryIntervalMs);
           this.retryTimer.unref();
         }
-        // If we are not exporting statsbeat and statsbeat is not disabled -- count success
+        // If we are not exportings statsbeat and statsbeat is not disabled -- count success
         if (!this.isStatsbeatSender) {
           this.networkStatsbeatMetrics?.countSuccess(duration);
+          for (const envelope of envelopes) {
```

Comment:
Is better to have this logic in statsbeat class, so all logic is available there and this code should be as simple as possible, in this case sending all the envelopes if needed to the method instead of iterating through here, also you are hardcoding the number 1 in number of envelopes, so this logic feels weird, maybe a refactor of the method is needed.